atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraTimestampStore.java

```diff
@@ -148,4 +200,32 @@ private static ConcurrentModificationException constructConcurrentTimestampStore
     private static String replaceBracesWithStringFormatSpecifier(String msg) {
         return msg.replaceAll("\\{\\}", "%s");
     }
+
+    private static CqlResult executeQueryUnchecked(Cassandra.Client client, ByteBuffer queryBuffer) {
+        try {
+            return client.execute_cql3_query(queryBuffer, Compression.NONE, ConsistencyLevel.QUORUM);
+        } catch (TException e) {
+            throw Throwables.rewrapAndThrowUncheckedException(e);
+        }
+    }
+
+    private static BoundData getCurrentBoundData(Cassandra.Client client) {
+        ByteBuffer selectQuery = CassandraTimestampUtils.constructSelectFromTimestampTableQuery();
+        CqlResult existingData = executeQueryUnchecked(client, selectQuery);
+        Map<String, byte[]> columnarResults = CassandraTimestampUtils.getValuesFromSelectionResult(existingData);
+
+        return ImmutableBoundData.builder()
+                .bound(columnarResults.get(CassandraTimestampUtils.ROW_AND_COLUMN_NAME))
+                .backupBound(columnarResults.get(CassandraTimestampUtils.BACKUP_COLUMN_NAME))
+                .build();
+    }
+
+    @Value.Immutable
+    interface BoundData {
```

Comment:
I thought the `@Value.Immutable` tag mainly went with abstract classes; why is an interface used here?